lib/Doctrine/DBAL/Schema/Table.php

```diff
@@ -634,11 +634,7 @@ public function getColumns()
         }
         $colNames = array_unique(array_merge($pkCols, $fkCols, array_keys($columns)));
 
-        uksort($columns, function ($a, $b) use ($colNames) {
-            return (array_search($a, $colNames) >= array_search($b, $colNames));
-        });
-
-        return $columns;
+        return array_replace(array_flip($colNames), $columns);
```

Comment:
I'm using php 5.6+ syntax since [composer.json](https://github.com/doctrine/dbal/blob/master/composer.json#L15) has
```
"require": {
        "php": "^7.0",
```